scripts/zkai_stwo_transformer_block_plan.py

```
@@ -0,0 +1,231 @@
+    baseline = _required_dict(plan.get("current_baseline"), "current_baseline")
+    if baseline.get("statement_receipt_schema") != "zkai-statement-receipt-v1":
+        raise PlanValidationError("current_baseline.statement_receipt_schema is not the checked receipt schema")
+    if baseline.get("proof_system") != "stwo-transparent-stark":
+        raise PlanValidationError("current_baseline.proof_system must stay Stwo-native")
+    if baseline.get("model_id") != "urn:zkai:ptvm:linear-block-v4-with-lookup":
+        raise PlanValidationError("current_baseline.model_id does not match the checked Stwo primitive")
+
+    mutation_result = _required_dict(baseline.get("mutation_result"), "current_baseline.mutation_result")
+    statement = _required_dict(mutation_result.get("statement_envelope"), "mutation_result.statement_envelope")
+    if statement.get("mutations_rejected") != statement.get("mutations_checked"):
+        raise PlanValidationError("current statement-envelope baseline must reject all checked mutations")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)